src/plugins.ts

```diff
@@ -0,0 +1,28 @@
+import { Rule } from './prefixer'
+
+export interface RuleSet {
+  selector: string,
+  rules: Rule[]
+}
+
+type Plugin = (set: RuleSet) => RuleSet
+
+const plugins: Plugin[] = []
+
+export function use(plugin: Plugin): void {
+  plugins.unshift(plugin)
+}
```

Comment:
We may want to decorate these functions with a little jsdoc description so we can easily understand the _why_ better. 
```suggestion
/**
 * Adds a plugin that ui-box will apply before emitting styles into a stylesheet.
 * Plugins are executed in the *opposite* order of insertion.
 */
export function use(plugin: Plugin): void {
  plugins.unshift(plugin)
}
```